metis-indexing/src/main/java/eu/europeana/indexing/IndexingConnectionProvider.java

```diff
@@ -0,0 +1,195 @@
+package eu.europeana.indexing;
+
+import java.io.Closeable;
+import java.net.InetSocketAddress;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.solr.client.solrj.SolrServer;
+import org.apache.solr.client.solrj.impl.CloudSolrServer;
+import org.apache.solr.client.solrj.impl.LBHttpSolrServer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientOptions;
+import com.mongodb.MongoClientOptions.Builder;
+import com.mongodb.MongoCredential;
+import com.mongodb.ServerAddress;
+import eu.europeana.corelib.edm.exceptions.MongoDBException;
+import eu.europeana.corelib.mongo.server.EdmMongoServer;
+import eu.europeana.corelib.mongo.server.impl.EdmMongoServerImpl;
+
+/**
+ * <p>
+ * This class is maintainable for providing an instance of {@link Indexer} with the required
+ * connections to persistence storage. It is responsible for maintaining the connections and
+ * providing access to the following functionality:
+ * <ol>
+ * <li>A DAO object (instance of {@link FullBeanDao}) for storing Full Beans.</li>
+ * <li>A Publisher object (instance of {@link FullBeanPublisher}) for publishing Full Beans to be
+ * accessed by external agents.</li>
+ * </ol>
+ * </p>
+ * <p>
+ * Please note that this class is {@link Closeable} and must be closed to release it's resources.
+ * </p>
+ * 
+ * @author jochen
+ *
+ */
+class IndexingConnectionProvider implements Closeable {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(IndexingConnectionProvider.class);
+
+  private final LBHttpSolrServer httpSolrServer;
+  private final CloudSolrServer zookeeperServer;
```

Comment:
Rename this to the same name as the class name(an the corresponding initializer methods)? It is actually a cloudSolrServer regardless that the zookeeper configuration is required when setting this up.

Reply:
Fixed. 

I thought it would be best however to not rename the static toZookeeperAddressString methods, as they refer to the Zookeeper documentation for specifying the acceptable format. What do you think?